mattermost-plugin/webapp/loaders/globalScssClassLoader.js

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+function blockList(line) {
+    return line.startsWith('.focalboard-body') ||
+        line.startsWith('.GlobalHeaderComponent') ||
+        line.startsWith('.channel-header__icon .LogoIcon') ||
+        line.startsWith('.focalboard-plugin-root');
+}
+
+module.exports = function loader(source) {
+    var newSource = [];
+    source.split('\n').forEach((line) => {
+        if (line.startsWith('.') && !blockList(line)) {
```

Comment:
What happens if there will be spaces at the line start like this for example:
```css
  .BlockIconSelector { /* two spaces at the line start */
    .octo-icon {
        display: flex;
        line-height: 1.1;
```
Are there any scss style checkers for this?

Reply:
It shouldn't happen if the scss is correctly formatted. We can add a linter for that.

I know, this is not the most robust approach but is less error prone than the existing one IMO.

Reply:
Yes, linter should help. May be [Stylelint can be used](https://stylelint.io/demo/#N4Igxg9gJgpiBcIAESB0AhANhMBrAkpAHYDKMmMYALhAE5LBID0AVElQO4RIDOADgEMwMHkgFV2ACxhJMASyIyeVAbQksmAHSIoUqHDQC0c4g226LSKHP6YBAT3hIAZhQAeAbnOWU8xYek5AHNJKicARlRwrx1dAF9tBKIQABoQHnsiFTcEdLAeHlTwCCJnYNzgb00QWgBXCh5qp0rYlGrxQzqKQyIIQ1qiXF6OIib2OpgU7zaQACNsPB6+mABbPip7MaoJqdakashsWiXjIgA3AXkoAJg3LZ3p-eKVlZgsk9X1zYRx2snH6qwMB2WjiOQlQzzHC4E5QWp8eRgcQwQxgWrKCArQx8WgQPgwNRyET3P67CyAyggsEQqGLXqGOEIkzI7G4-GE4k-ADajxQ21JvLMe3JIGCvVoMDGPOFlgOJR4lFqVDkZxRjMRyJ4hg4ciokgZcmczgJbyohgumD+jRAgpQAF1BUkLHayboKcDVNSiJCFjD6TxJHQ9QIiNccXiCRtDBBVbRaHJYNanPz-ntqs4SmbnAIVnJMPZYfCNVQUUQc5zkw80yAM+9s7n8ydcwUFEFDEE3gSTIZ63mC7gYPYuLQoCTUyLnANqODvUjMGATgoLQmx66ZpOiNOIXOF-SBvwhDBrhHQTRaKuATWp8qIX4YKp26DrKaTr0iMoQ1BVNdrBKt6MfhTNcngHexnFBV4GUpT0b29ek5DWIMQyoC9q1eawBB7e8qFqCUenLE4BiGCARlQkUy3Qx8E0MVR7y1eDzkuFdAKrci+kTYRQ1bQx+EoQ0TF1b5KwFat6XVZkSxos0EL4INOnqCtfnHN0QDEosJJRBUKGoOgkyU4DqnpT4ox4CBcOEMiVKMtxtkwhVc0OeVLJmBjlx-Mz5k0uwA1RTFXiyPSgMvVymLDCAeF1GcpOMRC1HkihnKeeC4xgIJ6gfDhJF1ERBAsliRJFcN2SjPdBmGADhOUmYtMoM9sQVWooD6D0CkIsqSIq-TLxqnTjj4BqmsMchVhfUriNI-KqqeHq6o2fE2vGzrpR8IKZRaHwZjFOhJW5W0VLRDEsWG-yqGtPaHRlJ1dBdbrtm4+lFA4O9EuqAZdQW8qXpADouhRYyCzvSEYAzCUpUFdpMA4BxrWAlB1o26pbmEdYwZlEVaVwBotQEZwS2OHgCIolEMax6pYdlGs5FoZQehEEtRxtNGkBupnqi20HdqZmYcbx3yXlNapzsdR4WZFH6FPwyCkQVMZqmwDgCTJy9xe6ImeNylEecVn4Iah+wtQioggm6Z7UmVs1fp41YTAgbA4JgJ6FE13HtacXXoaV6sMdRbBDbbWZQWEIa1ijQHZmB7bZdUmBY099HfR98LuIDw9acd-wtfPHXvshj2za9hPgSTo3IUD0sHbDiOObdnO9a1FZ6mVQxTYMuZC995Oy-V1Pw5Bnaa8uOueNbE2nbjlTvYjIhO9Tx7AczqPB+hwwG8wJuW8vSf8WnkuU6Dg8g4X7Ol-14ejdHxRx5mLe3hn-eNaBvvF9z-Wr6eRzjmkNxUQEGXs-l7WrcP43G-hQI2eoo4BiDG-OU-N3j-Wbk7R+kdOY+Hdq-fOPh4ZoJAEjGAKNUEbXXFTGmihlBHkFkzUWCNRRBHFP3JAy0iHTQ2BQPwZpIAvE-GdKhws9jUJmJwk62osolgPiiBQEVYDPzrjA8A6IaBYiKpGAsCDK5P0IWLF+MNBTYIpngghTgmFEPaC7Y4B1FGshPBsN+OCyjUzNGQ+mlCiECIpuzBhxiaGZz5idWx7j3wJk0iPFEYdfQuI2hdHwV17StyBFSWCkIQxtnETRMxUdFCx0wSpeJMEoqzGSd3IOvcUED20XI3Jp58kJ3sjbO2ad57pOPtolejc5CIMvtkmYlSvQ+mhFbBytsISpKPmUoefsL47TidBKpNIanW0ct6VJJTq5JRjoAy8PTEnewmaEpBKwBDfy2TOPS4Rpkelmd6b2tk8zcVqYsmRedzkJKioshpSDRlPBPvXNpHSpmbJmb0t5IymljOXrsv5FTAWJOBQ-FZDDDLrKzs8vJEI1FIPhajHB3z-F6JFAYlCmjvFmKghcr0-iJwkMcXTChjNXHk02nQ0pjC9rcxJUIgWXSaGSKCWfY2ez-AYwiT4KJlgYnM1bhuf8vjMJzw+aCr5LTV7rzHly9M15Xl+Tsg-T56CDYhMhWqq8m4YVaqKcTKuCLo5ZMlRqiEBzv7orIWMAADLak1UU1bSytQA5Fl4pWJMEBKLI0gFT0Qrkgnl0jmlD2Ve0je1YA1RSDaaUNIhzWnCkVazJBI+WTLkUmiEmVsogrxo8jBrcFCwCyOSn4AAWVuYCgh6kMAALwJH0UquovoOuDl8P5pzW7oTkJhY04hcIolhanXVtcnmXmHaO7CE7fL1OWZajJSK5ELqwuOvCXrf4+pIhstCR4R07pwnhFNIaRDptSVG7Nm6jXbrHRelEoIjYohPOIOgGaZ04qfaexdu633JM-eyb9+M4XrpjXOk9GFDAAEc-i0ABjYDhZq5UZwVXq1pa942qqHYBxDyHUM0yEdq6d2HZ2nwhQmkU26kMElI+hrhGbMXZxzX60SyxbYiJLRrL69Jmx+yGqGaMzgeJmVoEHTDDDVrkVqCscOxwKACGsCXdtuJy06NEop5TJwbl+DbJp8KX1lFqALN6qOvq5GWydRajRRjwbUZ+Xhw15M8UqQJVi5h6YqW03IQzc6DKngeJ8yY767K-KcqFpdEWrcZo-vEHdWYSpiaBwHKdDN96N02u6sNXqUkUtpejOBuqpbXZrLy9WRLxxkvxlS5JL95WoOOaq8ekUtW+azAUBB39VH-0JYK3VThPWywtZ7tBsFFb8vaTqhxN4X53ijd63VB6wyBOTSG3Nn98gyMYYjVhstMGZs1eG7ttDMq2NTfa1xzr53ji9vs3pN1s3ao-v6jARqzVvJais--I9d2VJdc+99n2v8tRXr1Deg2D8csccfdt97fUBrLAoMI-7NcbNGpB6joa6OXxvN6CoWCUcmq1E8nIrrc1J0Hus4DuRb0OF04BwrIHMwLR-EMKBYc1xMdPGx63Tnez9usdk2k4703XMqs6ULy4XO9ssYOf1yXirxkGroypYXiDRfK7XW1xF1WRTa8VyvQ5fbQ5Oxe94JIcQQBxCAA).

Reply:
What about ID selector?

Reply:
@jespino if we know adding a linter will make it more robust, then let's do it right away instead of waiting for something to break first.

Reply:
Yes, there is another PR to handled that here: #1078 

About the ID, I'm not super concern about that, because we are not using root `ID` selector in general, but I can add that too.